### ISSUE TITLE
Character count does not calculate correctly when using a Jinja placeholder in the template

### ISSUE THREAD

**Issue by ishiis** [sqlfluff/sqlfluff#5393]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

When I execute the `sqlfluff lint` command on SQL with a Jinja placeholder in the template, I encounter a bug.

I have set the `max_line_length` to 80. The actual SQL line length is 61 characters.
However, the command results in a line length of 91 characters.

The character count does not calculate correctly when using a Jinja placeholder in the template.

```bash
% cat example.sql
SELECT * FROM `{{ var.value.gcp_project_id }}.dataset.table`;

% head -1 example.sql | tr -d '\n' | wc -c
61

% sqlfluff lint example.sql
== [example.sql] FAIL
L: 1 | P: 1 | LT05 | Line is too long (91 > 80).
| [layout.long_lines]
All Finished 📜 🎉!
```

### Expected Behaviour

The expected behavior is not to fail if the number of characters in the line is less than `max_line_length`.

### Observed Behaviour

The observed behavior is that it fails even if the number of characters on the line is less than `max_line_length`.

### How to reproduce

Here are the steps to reproduce this issue:

```bash
% echo 'SELECT * FROM `{{ var.value.gcp_project_id }}.dataset.table`;' > example.sql
% sqlfluff lint example.sql
```

### Dialect

bigquery

### Version

```bash
% sqlfluff --version
sqlfluff, version 2.3.5
```

vscode-sqlfluff version is `2.4.4`.

### Configuration

```toml
[tool.sqlfluff.core]
dialect = "bigquery"
templater = "jinja"
sql_file_exts = ".sql,.sql.j2,.dml,.ddl"
max_line_length = 80

[tool.sqlfluff.indentation]
indented_joins = false
indented_using_on = true
template_blocks_indent = false

[tool.sqlfluff.templater]
unwrap_wrapped_queries = true

[tool.sqlfluff.templater.jinja.context]
var.value = "" # for apache airflow variables

[tool.sqlfluff.rules.capitalisation.keywords]
capitalisation_policy = "upper"
```

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by ishiis** [comment]
Fixing this issue is very difficult for me. However, if I had some hints, I might be able to create a Pull request.
Please let me know if there’s anything I can do.

**Comment by ishiis** [comment]
It’s possible that the placeholders in the Jinja template are being **counted twice**.

In Case2 adds 3 characters to the Jinja placeholder in Case1 SQL, it increases by 6 characters.
Case2 and Case3 are adding the same number of characters, but the results are different.

## Case1:

The character count for SQL is 61 characters.
The character count for the Jinja placeholder is 30 characters.
The total character count is 91 characters.

```bash
% echo 'SELECT * FROM `{{ var.value.gcp_project_id }}.dataset.table`;' > example.sql
% head -1 example.sql | tr -d '\n' | wc -c
61
% sqlfluff lint example.sql
== [example.sql] FAIL
L: 1 | P: 1 | LT05 | Line is too long (91 > 80).
| [layout.long_lines]
All Finished 📜 🎉!
```

## Case2:

I add 3 characters to the Jinja placeholder.
The number of characters in the SQL is 64.
The number of characters in the Jinja placeholder is 33.
The total number of characters is 97.

```bash
% echo 'SELECT * FROM `{{ var.value.gcp_project_id123 }}.dataset.table`;' > example.sql
% head -1 example.sql | tr -d '\n' | wc -c
64
% sqlfluff lint example.sql
== [example.sql] FAIL
L: 1 | P: 1 | LT05 | Line is too long (97 > 80).
| [layout.long_lines]
All Finished 📜 🎉!
```

## Case3:

I add 3 characters to the SQL that is not a Jinja placeholder.
The number of characters in the SQL is 64.
The number of characters in the Jinja placeholder is 30.
The total number of characters is 94.

```bash
% echo 'SELECT * FROM `{{ var.value.gcp_project_id }}.dataset.table123`;' > example.sql
% head -1 example.sql | tr -d '\n' | wc -c
64
% sqlfluff lint example.sql
== [example.sql] FAIL
L: 1 | P: 1 | LT05 | Line is too long (94 > 80).
| [layout.long_lines]
All Finished 📜 🎉!
```

**Comment by peterbud** [comment]
Hi @ishiis , 

There are multiple problems with your config above:
* In the context you have only `var.value`, but in the SQL you are looking for `var.value.gcp_project_id` - so there is actually no variable replaced. So you need to have the full variable in the context
* You have a variable of `""`, and the rendered `SELECT * FROM "".dataset.table;` is not a valid SQL in bigquery dialect. If you add `var.value.gcp_project_id = A` to the context, without the quotation marks, then it will work. Also do NOT att comments at the end of the variable line as it will be also rendered, and that is leading to the long line error
